File: Basic/Basic_Recursion/reverse_array.cpp
#include <iostream>
using namespace std;
void helper(vector<int> &arr, int low, int high)
{
    if (low >= high)
    {
        return;
    }
    swap(arr[low], arr[high]);
    helper(arr, low + 1, high - 1);
}
void reverseArray(vector<int> &arr)
{
    helper(arr, 0, arr.size() - 1);
}

File: Basic/Basic_Recursion/reverse_array.py
def helper(arr, low, high):
    if low >= high:
        return
    arr[low], arr[high] = arr[high], arr[low]
    helper(arr, low + 1, high - 1)


def reverse_array(arr):
    helper(arr, 0, len(arr) - 1)
    return arr
